product_state/models/product_template.py

```diff
@@ -1,18 +1,39 @@
 # Copyright 2017 ACSONE SA/NV (<http://acsone.eu>)
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
 
-from odoo import models, fields
+from odoo import api, models, fields
+
+
+class ProductState(models.Model):
+    _name = 'product.state'
+    _description = "Product State"
+    _order = 'name'
+
+    name = fields.Char('State Name', required=True)
+    description = fields.Text(translate=True)
+    product_ids = fields.One2many(
+        'product.template',
+        'product_state_id',
+        string='State Products',
```

Comment:
Thanks for you interesting contribution. I think there is an opportunity to add another improvement: put this field on product.product instead of product.template.
Because you may have only one variant that can be obsolete (pink tee-shirt) and not all tee-shirt. Yes it's not the object of this improvement but while doing a migration script, it also could be done.
What do you think ?

Reply:
Yes, I like it. I am focus right now in product.template but once I get it i will try to add the variants too